flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/StreamTableEnvironment.scala

```diff
@@ -555,18 +555,105 @@ abstract class StreamTableEnvironment(
           s"But is: ${execEnv.getStreamTimeCharacteristic}")
     }
 
+    // Can not apply key on append stream
+    if (extractUniqueKeys(fields).nonEmpty) {
+      throw new TableException(
+        s"Can not apply key on append stream, use fromUpsertStream instead.")
```

Comment:
Agree, specifing unique key on an AppendTable makes sense. For example, we only insert data into a table without an update. 
Since supporting keys on AppendStream is another new feature. I think we can add this check now and remove it later if we support keys on AppendStream. The error message should be corrected as `Apply key on append stream has not been supported yet! You can use fromUpsertStream instead.`
What do you think?

Reply:
I meant that this check doesn't prevent from any bugs or from any other exceptions. It's just to block user from specifying a no-op key definition, right? That's why I'm not sure if we should block it.

If we decide to keep this check/exception, I would refrain from naming `fromUpsertStream` in the exception. Maybe rephrasing it to:
> Defining key on append stream do not have any effects

?

Reply:
I haven't thought much about how to support key definition on append stream. I think it may not be a no-op key definition. Maybe we have to do the following things:
- Add similar logic in `UniqueKeyExtractor` to get keys.
- Probably have to check whether data meet the key constraint. i.e, throw an exception when key data is not distinct. 
- etc.
Given the above things. I think we would better to not support it now? 
I will rephrase the error message according to your suggestions. Thanks a lot.